lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -170,16 +179,16 @@ void FIRRTLTypesLowering::runOnOperation() {
     dispatchVisitor(&op);
   }
 
-  // Remove ops that have been lowered.
-  while (!opsToRemove.empty())
-    opsToRemove.pop_back_val()->erase();
+  // Remove ops that have been lowered. Erasing in reverse order so we don't
+  // have to worry about calling dropAllUses before deleting an operation.
+  for (auto *op : llvm::reverse(opsToRemove))
+    op->erase();
```

Comment:
Does something need to clear opsToRemove, or it get reinitialized?

Reply:
Now that we're re-recreating the Visitor for each module, it gets reinitialized.